src/com/company/servicii/CSVReadService.java

```diff
@@ -0,0 +1,201 @@
+package com.company.servicii;
+
+import com.company.entitati.Analiza;
+import com.company.entitati.Consultatie;
+import com.company.entitati.Ecografie;
+import com.company.entitati.Medic;
+
+import java.io.BufferedReader;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.IOException;
+import java.sql.Array;
+import java.util.*;
+
+public class CSVReadService {
+    private static CSVReadService instance;
+    public static synchronized CSVReadService getInstance() {
+        if(instance == null) {
+            instance = new CSVReadService();
+        }
+        return instance;
+    }
```

Comment:
no private constructor declared and in Main you should call getInstance method to get your singleton instance, not call the constructor as you do now for all the wannabe singleton services :D

Reply:
please fix this singleton issue with no private constructors in all classes that are intended to be singletons